Data/Vector/Storable.hs

```diff
@@ -980,6 +984,20 @@ izipWith6 :: (Storable a, Storable b, Storable c, Storable d, Storable e,
 {-# INLINE izipWith6 #-}
 izipWith6 = G.izipWith6
 
+-- | Faster equality comparison which uses pointer equality as
+-- optimization. Note it's not the same thing as @==@ when element
+-- type doesn't have lawful 'Eq' instance. For example:
+--
+-- >>> let xs = fromList [0/0::Double] in fastEq xs xs
+-- True
+--
+-- >>> let xs = fromList [0/0::Double] in xs == xs
+-- Falsec
+fastEq :: (Storable a, Eq a) => Vector a -> Vector a -> Bool
+fastEq v1@(Vector n1 ptr1) v2@(Vector n2 ptr2)
+  | n1 == n2 && ptr1 == ptr2 = True
+  | otherwise                = v1 == v2
```

Comment:
Could we rather expose 
```haskell
isSamePtr :: Vector a -> Vector a -> Bool
isSamePtr (Vector n1 ptr1) (Vector n2 ptr2) = n1 == n2 && ptr1 == ptr2
```
and let power users construct `fastEq` themselves?

This routine is faster only in a very specific case and is (marginally) slower otherwise. I'm not convinced this warrants `fastEq` name.

Reply:
That's good idea. It's probably better to call  it `isSameBuffer` since pointers could be same but length differ.

Reply:
I'd rather call it `isSameVector`. In order to know that it is a pointer or a buffer you need to understand the internals, but `isSameVector` will be clear from the name what  it means. More over different offsets do means different vectors, but do not mean that the buffer or the pointer are different.